server/controllers/user.js

```diff
@@ -0,0 +1,77 @@
+import userRepo from '../repository/userRepository';
+import generateToken from '../utilities/jwtGenerator';
+import passwordUtil from '../utilities/passwordHasher';
+import { goodHttpResponse, badHttpResponse } from '../utilities/httpResponse';
+
+const { hash } = passwordUtil;
+/**
+ * User Controller class
+ */
+class User {
+  /**
+   * Register new User
+   * @param {object} request Request Object
+   * @param {object} response Response Object
+   * @returns {object} User Object
+   */
+  static async signup(request, response) {
+    const {
+      firstName,
+      lastName,
+      email,
+      username,
+      password,
+    } = request.body;
+
+    const user = await userRepo.getUserByCredentials(email, username);
+    if (user) {
+      return badHttpResponse(
+        response,
+        409,
+        'This account already exists. Consider sign in.',
+        'We found another user with the same email and/or username'
+      );
+    }
+
+    const newUser = await userRepo.createUser({
+      firstName,
+      lastName,
+      email,
+      username,
+      password: hash(password),
+    });
+
+    const token = generateToken(newUser.id);
+    return goodHttpResponse(
+      response,
+      201,
+      `Hello ${newUser.username}, Welcome to Author's Haven.`,
+      { token }
+    );
+  }
+
+  /**
+   * Get all non-admin users.
+   * @param {object} request
+   * @param {object} response
+   * @returns {object} The JSON response to the user.
+   */
+  static async listAll(request, response) {
+    const allUsers = await userRepo.getAllUsers();
+    if (allUsers === null) {
+      return badHttpResponse(
+        response,
+        404,
```

Comment:
When users are not found, I don't think it is a bad request. It should be 200 but maybe with a message that says there are no users yet or something similar.